File: app/src/main/java/com/ravisharma/playbackmusic/database/PlaylistDao.java
package com.ravisharma.playbackmusic.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import com.ravisharma.playbackmusic.model.Playlist;

import java.util.List;

@Dao
public interface PlaylistDao {

    @Insert
    void addSong(Playlist playlist);

    @Query("DELETE FROM playlistTable WHERE id=:songId")
    void removeSong(long songId);

    @Query("DELETE FROM playlistTable WHERE playlistName=:playlistName AND id=:songId")
    void removeSong(String playlistName, long songId);

    @Query("DELETE FROM playlistTable WHERE playlistName=:playlistName")
    void removePlaylist(String playlistName);


    @Query("SELECT * FROM playlistTable")
    List<Playlist> getAllPlaylistSongs();

    @Query("SELECT * FROM playlistTable WHERE playlistName=:playlistName")
    LiveData<List<Playlist>> getPlaylistSong(String playlistName);

    @Query("SELECT * FROM playlistTable WHERE playlistName=:playlistName")
    List<Playlist> getPlaylist(String playlistName);

    @Query("SELECT COUNT(*) FROM playlistTable WHERE playlistName=:playlistName AND id=:songId")
    long isSongExist(String playlistName, long songId);
}

File: app/src/main/java/com/ravisharma/playbackmusic/database/dao/PlaylistDao.kt
package com.ravisharma.playbackmusic.database.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.ravisharma.playbackmusic.model.Playlist

@Dao
interface PlaylistDao {
    @Insert
    fun addSong(playlist: Playlist)

    @Query("DELETE FROM playlistTable WHERE id=:songId")
    fun removeSong(songId: Long)

    @Query("DELETE FROM playlistTable WHERE playlistName=:playlistName AND id=:songId")
    fun removeSong(playlistName: String, songId: Long)

    @Query("DELETE FROM playlistTable WHERE playlistName=:playlistName")
    fun removePlaylist(playlistName: String)

    @Query("UPDATE playlistTable SET playlistName=:newPlaylistName WHERE playlistName=:oldPlaylistName")
    suspend fun renamePlaylist(oldPlaylistName: String, newPlaylistName: String)

    @get:Query("SELECT * FROM playlistTable")
    val allPlaylistSongs: List<Playlist>

    @Query("SELECT * FROM playlistTable WHERE playlistName=:playlistName")
    fun getPlaylistSong(playlistName: String): LiveData<List<Playlist>>

    @Query("SELECT * FROM playlistTable WHERE playlistName=:playlistName")
    fun getPlaylist(playlistName: String): List<Playlist>

    @Query("SELECT COUNT(*) FROM playlistTable WHERE playlistName=:playlistName AND id=:songId")
    fun isSongExist(playlistName: String, songId: Long): Long
}